Lib/asyncio/unix_events.py

```diff
@@ -166,7 +172,13 @@ def remove_signal_handler(self, sig):
 
         if not self._signal_handlers:
             try:
-                signal.set_wakeup_fd(-1)
+                oldfd = signal.set_wakeup_fd(-1)
+                if oldfd != -1 and oldfd != self._csock.fileno():
+                    warnings.warn(
+                        'Got unexpected signal wakeup file '
+                        'descriptor while removing it',
+                        ResourceWarning,
+                        source=self)
```

Comment:
I don't really like warnings being issued when closing things -- they aren't very actionable. I understand this is trying to alert the user that something *else* might have set the wakeup fd, overriding ours. But if we made it this far there's not much we can do about it, is there?

(OTOH maybe this could be logged in debug mode?)

Reply:
> But if we made it this far there's not much we can do about it, is there?

That's fair, though I don't expect the user to act on the warning programmatically, even if they could. It's a hint.

> OTOH maybe this could be logged in debug mode?

I think that reduces the value of the hint. I would like to know that my application is potentially broken ASAP, rather than waiting for it to manifest itself somehow to even consider looking at debug logs.

Reply:
Fair enough. I'll hold off for a little bit and then merge.